### PR TITLE
feat(api): send errors to Sentry

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Key points:

- Sends errors when the DSN exists (oops)
- Uses the same response for all unhandled errors
- Doesn't mess with the weirder responses (I'm looking at you, donation and exams)

### CODE REVIEW

File: api/src/routes/user.ts [hunk 1]
```diff
@@ -231,44 +201,33 @@ export const userRoutes: FastifyPluginCallbackTypebox = (
...
-        return {
-          type: 'info',
-          message: 'flash.report-sent',
-          variables: { email: user.email }
-        } as const;
-      } catch (err) {
-        fastify.log.error(err);
-        // TODO: redirect to the reported user's profile if there's an error
-        void reply.code(500);
+      if (!username || !report || report === '') {
```
Comment (ShaunSHamilton) [inline api/src/routes/user.ts:209]: `"" == false`

```suggestion
      if (!username || !report) {
```

**Review by ShaunSHamilton — APPROVED** [review]
Two of those diffs are painful to look at. Overall, red number is bigger than green number 👍

**Comment by ojeytonwilliams** [comment]
Yeah, I know. They're just the result of removing try-catch blocks, but they're hard to read.

**Review by raisedadead — APPROVED** [review]

**Comment by raisedadead** [comment]
At some point I would love to move to OTEL collector instead of the Sentry SDK.

**Pull request merged**